robolectric-resources/src/main/java/org/robolectric/res/Fs.java

```diff
@@ -25,6 +26,13 @@
 abstract public class Fs {
   public static Fs fromJar(URL url) {
     try {
+      if (url.toURI().getPath().contains(" ")) {
+        try {
+          url = new URL(url.toURI().getPath().replace(" ", "%20"));
+        } catch (MalformedURLException e) {
+          e.printStackTrace();
```

Comment:
Shouldn't this throw IllegalArgumentException (as below)? That means you can also collapse the catch to the one below.